### PR TITLE
[DOCS] Revamps the Anomaly Detection section

### PR DESCRIPTION
**Pull request by szabosteve**
This PR reorganizes the ML Anomaly detection section to align with the ML feature template.

Related issue: https://github.com/elastic/stack-docs/issues/459

### CODE REVIEW

**Comment by lcawl** [comment]
FYI the Stack Overview now builds successfully, but when I build all docs, it generates the following:
INFO:build_docs:Bad cross-document links:
INFO:build_docs:  /tmp/docsbuild/target_repo/html/en/elastic-stack-overview/master/create-jobs.html:
INFO:build_docs:   - en/elastic-stack-overview/master/ml-configuring-pop.html
INFO:build_docs:  /tmp/docsbuild/target_repo/html/en/elasticsearch/reference/master/ml-datafeed-resource.html:
INFO:build_docs:   - en/elastic-stack-overview/master/ml-configuring-aggregation.html
INFO:build_docs:   - en/elastic-stack-overview/master/ml-configuring-transform.html
INFO:build_docs:   - en/elastic-stack-overview/master/ml-delayed-data-detection.html
INFO:build_docs:  /tmp/docsbuild/target_repo/html/en/elasticsearch/reference/master/ml-get-category.html:
INFO:build_docs:   - en/elastic-stack-overview/master/ml-configuring-categories.html
INFO:build_docs:  /tmp/docsbuild/target_repo/html/en/elasticsearch/reference/master/ml-job-resource.html:
INFO:build_docs:   - en/elastic-stack-overview/master/ml-configuring-categories.html
INFO:build_docs:   - en/elastic-stack-overview/master/ml-configuring-detector-custom-rules.html
INFO:build_docs:   - en/elastic-stack-overview/master/ml-configuring-pop.html
INFO:build_docs:   - en/elastic-stack-overview/master/ml-configuring-url.html
INFO:build_docs:  /tmp/docsbuild/target_repo/html/en/elasticsearch/reference/master/ml-results-resource.html:
INFO:build_docs:   - en/elastic-stack-overview/master/ml-configuring-categories.html
INFO:build_docs:  /tmp/docsbuild/target_repo/html/en/kibana/master/xpack-ml.html:
INFO:build_docs:   - en/elastic-stack-overview/master/ml-delayed-data-detection.html

Those will need to be addressed if this PR goes ahead.

File: docs/en/stack/ml/anomaly-detection/index.asciidoc [hunk 1]
```diff
@@ -4,37 +4,30 @@
...
+* <<ml-configuration>>
 //* <<ml-getting-started>>
-* <<ml-configuring>>
-* <<stopping-ml>>
-* <<ml-troubleshooting, Troubleshooting Machine Learning>>
-* <<ml-api-quickref>>
-* <<ml-functions>>
+* <<ml-reference>>
+* <<ml-limitations>>
+//* <<ml-troubleshooting>>
```
Comment (lcawl) [inline docs/en/stack/ml/anomaly-detection/index.asciidoc]: What's the plan for relocating this troubleshooting content?

Comment (szabosteve) [reply docs/en/stack/ml/anomaly-detection/index.asciidoc]: @lcawl AFAIK, the content under troubleshooting is not valid anymore. I'd like to double-check them and delete the irrelevant ones from the master/7.x/7.4 branches.

**Comment by szabosteve** [comment]
@elasticmachine update branch

**Comment by szabosteve** [comment]
run elasticsearch-ci/doc skiplinkcheck

**Comment by szabosteve** [comment]
run elasticsearch-ci/doc

**Comment by szabosteve** [comment]
@lcawl I changed the structure as we discussed.

Preview: http://stack-docs_548.docs-preview.app.elstc.co/guide/en/elastic-stack-overview/master/xpack-ml.html

**Comment by szabosteve** [comment]
Backported to 7.x, 7.4, and 7.3.

**Pull request merged**